### PR TITLE
Allows users to download reports on bedspace usage and occupancy

### PR DESCRIPTION
**Pull request by patrickjfl**
### Before

Screenshot 2023-05-11 at 14 55 53

### After

Screenshot 2023-05-11 at 14 25 13

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

File: server/@types/ui/index.d.ts [hunk 1]
```diff
@@ -269,3 +269,5 @@ export interface SideNavObj {
 }
 
 export type BookingSearchApiStatus = 'provisional' | 'confirmed' | 'arrived' | 'departed' | 'closed'
+
```
Comment (richpjames) [inline server/@types/ui/index.d.ts:272]: I think this change (updating the types) should be in the previous commit

File: server/controllers/temporary-accommodation/manage/reportsController.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ export default class ReportsController {
           reportType,
         )
       } catch (err) {
-        catchValidationErrorOrPropogate(req, res, err, paths.reports.bookings.new({}))
+        catchValidationErrorOrPropogate(req, res, err, paths.reports.new({}))
```
Comment (richpjames) [inline server/controllers/temporary-accommodation/manage/reportsController.ts:66]: I think this change should be in one of the earlier commits

File: integration_tests/tests/temporary-accommodation/manage/report.cy.ts [hunk 3]
```diff
@@ -73,25 +75,126 @@ context('Report', () => {
...
+
+    // When I visit the report page
+    cy.task('stubReportReferenceData')
+    const page = ReportNewPage.visit()
+
+    cy.then(function _() {
+      // Then I should see the user's probation region preselected
+      page.shouldPreselectProbationRegion(this.actingUserProbationRegion)
+
+      // And when I fill out the form
```
Comment (richpjames) [inline integration_tests/tests/temporary-accommodation/manage/report.cy.ts]: Might just be me but I'm not sure about this 'and when'  
I think we should have another given, when, then. 
Maybe something like 'Given I need a report, when I fill out...'

**Review by richpjames — APPROVED** [review]
Looks good to me, just a couple of minor comments

File: server/@types/ui/index.d.ts (hunk 1 above)
Comment (patrickjfl) [reply server/@types/ui/index.d.ts:272]: oh yeah, good spot!

File: server/controllers/temporary-accommodation/manage/reportsController.ts (hunk 2 above)
Comment (patrickjfl) [reply server/controllers/temporary-accommodation/manage/reportsController.ts:66]: yes, thank you

File: integration_tests/tests/temporary-accommodation/manage/report.cy.ts (hunk 3 above)
Comment (patrickjfl) [reply integration_tests/tests/temporary-accommodation/manage/report.cy.ts]: yeah I agree. I've tidied it up a bit

**Pull request merged**